[mlir][Interfaces] `ValueBoundsConstraintSet`: Add `dump` helper function

matthias-springer opened the pull request:
This commit adds a helper function that dumps the constraint set and the mapping of columns to values/dims. For debugging only.

Example output:
```
==========
Columns:
(column	dim	value)
 0	1	linalg.fill (result 0)
 1	1	tensor.extract_slice (result 0)
 2	n/a	affine.min (result 0)
 3	n/a	scf.for (bbarg 0)
 4	n/a	func.func (bbarg 2)

Constraint set:
Domain: 0, Range: 1, Symbols: 4, Locals: 0
6 constraints
(None	None	None	None	None	const)
 1	-1	0	0	0	0	= 0
 0	1	-1	0	0	0	= 0
 0	0	-1	-1	1	0	>= 0
 0	0	-1	0	0	4	>= 0
 0	0	0	1	0	0	>= 0
 0	0	0	-1	1	-1	>= 0
==========
```

MacDue reviewed (approved):
Thanks! This is very handy. Just one nit:

matthias-springer committed:
[mlir][Interfaces] `ValueBoundsConstraintSet`: Add `dump` helper function

This commit adds a helper function that dumps the constraint set and the mapping of columns to values/dims. For debugging only.

Example output:
```

Columns:
column	dim	value owner
0	1	linalg.fill
1	1	tensor.extract_slice
2	n/a	affine.min
3	n/a	scf.for
4	n/a	func.func

Constraint set:
Domain: 0, Range: 1, Symbols: 4, Locals: 0
6 constraints
(None	None	None	None	None	const)
 1	-1	0	0	0	0	= 0
 0	1	-1	0	0	0	= 0
 0	0	-1	-1	1	0	>= 0
 0	0	-1	0	0	4	>= 0
 0	0	0	1	0	0	>= 0
 0	0	0	-1	1	-1	>= 0

```
